portal-bridge/src/api/execution.rs

```diff
@@ -217,20 +218,26 @@ impl ExecutionApi {
     }
 
     /// Return unvalidated receipts for the given transaction hashes.
-    async fn get_trusted_receipts(&self, tx_hashes: &[B256]) -> anyhow::Result<Receipts> {
-        let request: Vec<JsonRequest> = tx_hashes
-            .iter()
-            .enumerate()
-            .map(|(id, tx_hash)| {
-                let tx_hash = hex_encode(tx_hash);
-                let params = Params::Array(vec![json!(tx_hash)]);
-                let method = "eth_getTransactionReceipt".to_string();
-                JsonRequest::new(method, params, id as u32)
-            })
-            .collect();
-        let response = self.batch_requests(request).await?;
-        serde_json::from_str(&response)
-            .map_err(|err| anyhow!("Unable to parse receipts from provider response: {err:?}"))
+    async fn get_trusted_receipts(&self, height: u64) -> anyhow::Result<Receipts> {
+        let block_param = format!("0x{height:01X}");
+        for _ in 0..3 {
```

Comment:
I'm not sure if this is the right place to implement a "retry" mechanism. If we do want this 3x retry mechanism, then it makes more sense to me to implement this inside `try_request()` - so that all functions that use `try_request()` will use it. But inside `try_request()` we already have a "fallback" attempt to the fallback provider, so I'm not even sure that this 3x retry mechanism is necessary?

Reply:
I don't think this retry method should be used in `try_request()`. I added it because of the specific constraints of `eth_getBlockReceipts`. At the moment I don't have the bandwidth or think this is high enough priority to think of the implications of implementing this suggestion.

I don't know if this retry mechanism would work for `get_block_number` or `get_header` and I don't think it should be as well.

The reason for this retry mechanism, is because we only call it when we *know* receipts should be available. In the case of fetching a latest header, this assumption isn't valid. I am also not aware if `get_block_number` etc use the same response pattern.

Reply:
Sometimes a receipt should be a avaliable but is still process, which can cause `getReceipts to return none`, which we seen in our method used before this PR

Reply:
Ok gotcha, then if you don't mind just adding a quick comment explaining why we have a dedicated retry for receipts only, since it's a special case where sometimes they're not returned when we expect them to be available